driver-legacy/src/main/com/mongodb/DBCollection.java

```diff
@@ -2195,6 +2201,19 @@ BsonDocument wrap(final DBObject document, @Nullable final Encoder<DBObject> enc
         }
     }
 
+    private ClientSideOperationTimeout getClientSideOperationTimeout(){
+       return ClientSideOperationTimeouts.create(database.getTimeoutMS());
```

Comment:
In the spec, there are overrides (Client, DB, Collection, and then (operation or ClientSession)). This seems to be handled by passing the database setting to the collection, allowing for an override via withTimeout. I think JAVA-4064 covers adding it to the client. Is there a test for what is currently implemented, or a Jira that covers whatever is left to finish and test?

Reply:
I'm expecting there to be Unified tests for timeouts. 
I haven't added tests as this PR is really to encapsulate what we currently have. But as you spotted in the MongoOperationPublisher I already hid a bug. Will review / ticket.